### PR TITLE
chore: troubleshooting buttons

### PR DESCRIPTION
**Pull request by deboer-tim**
### What does this PR do?

Updating all of the buttons in troubleshooting to our own button component.

With the updated design buttons don't show the 'in progress' spinner unless they have icons, so I added some generic fa icons. I also removed a couple titles (tooltips) that were identical to the button text, simplified some styling using space-x-, removing divs that are no longer needed, etc.

### Screenshot/screencast of this PR

https://github.com/containers/podman-desktop/assets/19958075/b9848282-a02f-495b-a84e-613bbf554674

### What issues does this PR fix or reference?

N/A

### How to test this PR?

Open troubleshooting page, click on buttons and check for styling issues.

### CODE REVIEW

File: packages/renderer/src/lib/troubleshooting/TroubleshootingContainerEngineGrabContainers.svelte [hunk 1]
```diff
@@ -28,18 +30,18 @@ async function grabContainers() {
...
+      icon="{faSignal}">
       Check containers
-    </button>
+    </Button>
   </div>
-  <div role="status" class="mx-2">{listContainersResult}</div>
+  <div role="status">{listContainersResult}</div>
   {#if listError}
-    <ErrorMessage class="mx-2" error="{listError}" />
+    <ErrorMessage error="{listError}" />
```
Comment (benoitf) [inline packages/renderer/src/lib/troubleshooting/TroubleshootingContainerEngineGrabContainers.svelte]: it's not related to buttons --> another PR please

File: packages/renderer/src/lib/troubleshooting/TroubleshootingContainerEngineGrabContainers.svelte [hunk 2]
```diff
@@ -28,18 +30,18 @@ async function grabContainers() {
...
+    <Button
+      bind:inProgress="{listInProgress}"
+      class="my-1 w-full"
+      on:click="{() => grabContainers()}"
+      icon="{faSignal}">
       Check containers
-    </button>
+    </Button>
   </div>
-  <div role="status" class="mx-2">{listContainersResult}</div>
```
Comment (benoitf) [inline packages/renderer/src/lib/troubleshooting/TroubleshootingContainerEngineGrabContainers.svelte:41]: it's not related to buttons --> another PR please

File: packages/renderer/src/lib/troubleshooting/TroubleshootingContainerEnginePing.svelte [hunk 3]
```diff
@@ -29,18 +31,18 @@ async function pingConnection() {
...
+      icon="{faSignal}">
       Ping
-    </button>
+    </Button>
   </div>
-  <div role="status" class="mx-2">{pingResult}</div>
+  <div role="status">{pingResult}</div>
   {#if pingError}
-    <ErrorMessage class="mx-2" error="{pingError}" />
+    <ErrorMessage error="{pingError}" />
```
Comment (benoitf) [inline packages/renderer/src/lib/troubleshooting/TroubleshootingContainerEnginePing.svelte]: it's not related to buttons --> another PR please

File: packages/renderer/src/lib/troubleshooting/TroubleshootingContainerEngineReconnect.svelte [hunk 4]
```diff
@@ -23,16 +25,16 @@ async function reconnectContainerProviders() {
...
+    on:click="{() => reconnectContainerProviders()}"
+    icon="{faPlug}">
     Reconnect providers
-  </button>
-  <div role="status" class="mx-2">{reconnectResult}</div>
+  </Button>
+  <div role="status">{reconnectResult}</div>
   {#if reconnectError}
-    <ErrorMessage class="mx-2" error="{reconnectError}" />
+    <ErrorMessage error="{reconnectError}" />
```
Comment (benoitf) [inline packages/renderer/src/lib/troubleshooting/TroubleshootingContainerEngineReconnect.svelte]: it's not related to buttons --> another PR please

File: packages/renderer/src/lib/troubleshooting/TroubleshootingPageStore.svelte [hunk 5]
```diff
@@ -19,27 +21,23 @@ let openDetails = false;
...
-      class="underline outline-none"
-      title="Open Details"
-      aria-label="Open Details"
-      on:click="{() => (openDetails = true)}">
-      {eventStoreInfo.name}
-    </button>
-  </div>
+  <Button
+    bind:disabled="{fetchInProgress}"
+    class="underline"
```
Comment (benoitf) [inline packages/renderer/src/lib/troubleshooting/TroubleshootingPageStore.svelte]: not sure why we need that if we have type=link ?

shouldn't it already formatted correctly ?

File: packages/renderer/src/lib/troubleshooting/TroubleshootingPageStoreDetails.svelte [hunk 6]
```diff
@@ -39,33 +41,24 @@ async function fetch(): Promise<void> {
...
-              <button
-                disabled="{fetchInProgress}"
-                class="px-3 my-1 text-sm font-medium text-center text-white bg-violet-600 rounded-sm hover:bg-dustypurple-800 focus:ring-2 focus:outline-none focus:ring-dustypurple-700"
-                title="Refresh"
-                on:click="{() => fetch()}">
-                Refresh
-              </button>
-            </div>
+          <div class="mx-2 flex flex-row items-center space-x-2">
+            Size:&nbsp;
```
Comment (benoitf) [inline packages/renderer/src/lib/troubleshooting/TroubleshootingPageStoreDetails.svelte]: why &nbsp; instead of using the mx-2 ?
also it should be part of another PR as it's not related to buttons

File: packages/renderer/src/lib/troubleshooting/TroubleshootingPageStoreDetails.svelte [hunk 7]
```diff
@@ -39,33 +41,24 @@ async function fetch(): Promise<void> {
...
           </div>
         </div>
 
         <!-- Display buffer events-->
         <div class="pb-4">
           <div class="bg-charcoal-800 w-full p-4 h-45">
-            <div class="pb-2 text-lg">
-              Updating events:
+            <div class="pb-2 text-lg space-x-2 my-1">
+              <span>Updating events:</span>
```
Comment (benoitf) [inline packages/renderer/src/lib/troubleshooting/TroubleshootingPageStoreDetails.svelte]: not related to buttons --> another PR please

File: packages/renderer/src/lib/troubleshooting/TroubleshootingPageStoreDetails.svelte [hunk 8]
```diff
@@ -39,33 +41,24 @@ async function fetch(): Promise<void> {
...
-                <button
-                  disabled="{fetchInProgress}"
-                  class="px-3 my-1 text-sm font-medium text-center text-white bg-violet-600 rounded-sm hover:bg-dustypurple-800 focus:ring-2 focus:outline-none focus:ring-dustypurple-700"
-                  title="Refresh"
-                  on:click="{() => eventStoreInfo.clearEvents()}">
-                  Clear events
-                </button>
+                <Button
+                  bind:inProgress="{fetchInProgress}"
+                  title="Clear events"
```
Comment (benoitf) [inline packages/renderer/src/lib/troubleshooting/TroubleshootingPageStoreDetails.svelte:62]: good catch !

**Review by benoitf — COMMENTED** [review]
I would not mix buttons change and layout changes as it makes review more complicated

File: packages/renderer/src/lib/troubleshooting/TroubleshootingContainerEngineGrabContainers.svelte (hunk 1 above)
Comment (deboer-tim) [reply packages/renderer/src/lib/troubleshooting/TroubleshootingContainerEngineGrabContainers.svelte]: Ok, will remove (repeat for all identical comments).

File: packages/renderer/src/lib/troubleshooting/TroubleshootingPageStore.svelte (hunk 5 above)
Comment (deboer-tim) [reply packages/renderer/src/lib/troubleshooting/TroubleshootingPageStore.svelte]: I didn't know these titles were clickable until I saw the code, and updating to our Button design meant the underline was only visible on hover which made it worse. So, I thought I would at least leave the underline for now.

I talked to Mairin briefly and she had some ideas how these links could be redesigned, but these are the one button on the page that aren't directly using PF classes nor copying the same design, so the simplest answer for this PR is to leave them as-is and handle another time. I've removed the changes to this 'button' in an updated commit.

File: packages/renderer/src/lib/troubleshooting/TroubleshootingContainerEngineGrabContainers.svelte (hunk 1 above)
Comment (deboer-tim) [reply packages/renderer/src/lib/troubleshooting/TroubleshootingContainerEngineGrabContainers.svelte]: Removed these changes and force-pushed a new commit.

File: packages/renderer/src/lib/troubleshooting/TroubleshootingPageStoreDetails.svelte (hunk 6 above)
Comment (deboer-tim) [reply packages/renderer/src/lib/troubleshooting/TroubleshootingPageStoreDetails.svelte]: I assume the character I can't see is a `nbsp;`? ;-)
It was showing up as`Size:1` without a space so I was just forcing one in there. Since it is not related to buttons I removed in new commit though.

**Review by cdrage — APPROVED** [review]
I don't see any styling issue with the latest update. The code LGTM (it's great we are now using the Button class), makes it wayyyy easier to comprehend.

LGTM!

File: packages/renderer/src/lib/troubleshooting/TroubleshootingContainerEngineReconnect.svelte [hunk 9]
```diff
@@ -24,13 +26,14 @@ async function reconnectContainerProviders() {
...
 <div class="flex flex-row items-center">
-  <button
-    disabled="{reconnectInProgress}"
-    class="px-3 my-1 text-sm font-medium text-center text-white bg-violet-600 rounded-sm hover:bg-dustypurple-800 focus:ring-2 focus:outline-none focus:ring-dustypurple-700"
-    title="Establish again connection to the socket of container engines"
-    on:click="{() => reconnectContainerProviders()}">
+  <Button
+    class="my-1"
+    bind:inProgress="{reconnectInProgress}"
+    title="Re-establish connection to the container engine sockets"
```
Comment (benoitf) [inline packages/renderer/src/lib/troubleshooting/TroubleshootingContainerEngineReconnect.svelte:32]: Thanks for the better wording 🎉

**Review by benoitf — APPROVED** [review]
thanks @deboer-tim 👍
